core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -324,15 +329,20 @@ private void initMetrics() {
                 .build();
     }
 
-    private void initStringSize(Session session) {
-        String configMaxStringSize = configurationService.load("org.hawkular.metrics", "string-size").toBlocking()
+    private void initConfiguration(Session session) {
+        Configuration configuration = configurationService.load("org.hawkular.metrics").toBlocking()
                 .lastOrDefault(null);
+        String configMaxStringSize = configuration.get("string-size");
         if (configMaxStringSize == null) {
             maxStringSize = -1;  // no size limit
         } else {
             maxStringSize = Integer.parseInt(configMaxStringSize);
         }
         log.infoMaxSizeStringMetrics(this.maxStringSize);
+
+        insertRetryMaxDelay = Long.parseLong(configuration.get("ingestion.retry.max-delay", "30000"));
```

Comment:
I do not think this logic is good because the value from the command line is overwritten with the value from the database every single time and there is no way to save value from the command line back into the db. I am missing something?